code/modules/mob/living/simple_animal/bot/hygienebot.dm

```diff
@@ -122,27 +115,29 @@
 				switch(frustration)
 					if(0 to 4)
 						currentspeed = 5
-						stop_washing()
 						mad = FALSE
-					if(4 to INFINITY)
+					if(5 to INFINITY)
 						currentspeed = 2.5
-						start_washing()
 						mad = TRUE
-			if(target)
+			if(target && !check_purity(target))
 				if(target.loc == loc && isturf(target.loc)) //LADIES AND GENTLEMAN WE GOTEM PREPARE TO DUMP
 					start_washing()
 					if(mad)
-						speak("Well about fucking time you degenerate", "Fucking finally", "Thank god, you finally stopped")
+						speak("Well about fucking time you degenerate.", "Fucking finally.", "Thank god, you finally stopped.")
 						playsound(loc, 'sound/effects/hygienebot_angry.ogg', 60, 1)
 						mad = FALSE
 					mode = BOT_SHOWERSTANCE
 				else
-					var/turf/olddist = get_dist(src, target)
+					stop_washing()
+					var/olddist = get_dist(src, target)
+					if(olddist > 7) // Target got away, focus on something else
+						back_to_idle()
+						return
 					walk_to(src, target,0, currentspeed)
-					if(mad && prob(60))
+					if(mad && prob(min(frustration, 60)))
```

Comment:
that's super low, can you add a * 2 or * 2.5 to frustration?

Reply:
I don't think it's super low. The chance is per tick, which is two seconds I believe? Starting at 5% and adding 1% every 2 seconds is high enough, especially if there's multiple.

Reply:
you need to consider that these changes stack up together though, they're (although only slightly) harder to build, and no longer chase for ages. This should already lower the amount of bots talking. Besides that them being mad should be them continiously ranting at you as you run, that's kind of the gimmick. 5% every life() is essentialy guaranteeing it'll only happen super rarely.